### PR TITLE
feat: allow rest-server to be mounted on a path to express

### PR DESCRIPTION
**Pull request by raymondfeng**
See https://github.com/strongloop/loopback-next/pull/2306#discussion_r252321716

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: greenkeeper.json [hunk 1]
```diff
@@ -33,5 +33,6 @@
         "sandbox/example/package.json"
       ]
     }
-  }
+  },
+  "ignore": ["@types/node"]
```
Comment (bajtos) [inline greenkeeper.json]: > chore: ignore @types/node upgrade by greenkeeper

What is the reasoning for this exclusion? Can you move this commit to a new pull request please? I don't see how it's related to RestServer improvements.

File: packages/rest/src/__tests__/integration/express.integration.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  function givenControllerClass(
+    // tslint:disable-next-line:no-any
+    ctor: new (...args: any[]) => Object,
+  ) {
+    server.controller(ctor);
+  }
+
+  function givenClient() {
+    const app = express();
```
Comment (bajtos) [inline packages/rest/src/__tests__/integration/express.integration.ts]: I find this very confusing. The function name says that it's creating a client, but it is creating an Express server too. 

I am proposing to rework the suite setup as follows:

```ts
beforeEach(givenLoopBackApp);
beforeEach(givenExpressServer);
beforeEach(givenClient);
```

File: greenkeeper.json (hunk 1 above)
Comment (raymondfeng) [reply greenkeeper.json]: Sorry, I guess I accidentally created this branch from the wrong base.

**Comment by raymondfeng** [comment]
@bajtos PTAL.

File: packages/rest/src/__tests__/integration/express.integration.ts (hunk 2 above)
Comment (raymondfeng) [reply packages/rest/src/__tests__/integration/express.integration.ts]: Fixed.

**Pull request merged**